codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithydafny/DafnyApiCodegen.java

```diff
@@ -87,8 +87,16 @@ public Map<Path, TokenTree> generate() {
           // Sort by shape ID for deterministic generated code
           .collect(Collectors.toCollection(TreeSet::new))
           .stream()
-          .map(this::generateCodeForShape)
-          .flatMap(Optional::stream)
+          .flatMap(shape -> {
+            final Optional<TokenTree> tokens = generateCodeForShape(shape);
+            if (tokens.isEmpty()) {
+              LOGGER.info(
```

Comment:
@alex-chew Intrigued by this - I feel like either it should be a warning, if we think this points to a degenerate model or likely internal bug or something like that, or should not be output at all, if sometimes it's perfectly reasonable to emit no code for some shapes, and outputting this might cause users to wonder if it's a problem.

My inclination is towards the latter. We could always add a -v verbose mode that outputs these kinds of messages for debugging.

Reply:
Yep, it seems reasonable to only output this in a verbose mode. I think I probably originally included this to aid in debugging.